### ISSUE TITLE
Rework peers identify to use the swarm closure

### ISSUE THREAD

**Issue by tomaka** [libp2p/rust-libp2p#294]
Right now the user has to handle incoming `identify` requests through the swarm closure, and can perform outgoing `identify` requests by using `swarm_controller.dial()`.

However the `PeerIdTransport` doesn't use this mechanism, and performs the initialization on his side instead. The results are passed as the `Transport` output.

I feel like the identification system should somehow be reworked in order to unify these two mechanisms.

**Comment by tomaka** [comment]
No longer relevant after #565
